tests/tests.wake

```diff
@@ -77,6 +87,77 @@ export def runTests (cmdline: List String): Result String Error =
 
     Pass "All {len tests | str} tests completed successfully."
 
+
+export def runUnitTests _: Result Unit Error =
+    def testDirectory = "{here}/wake-unit"
+    require Pass testScript =
+      source "{testDirectory}/unit-test.sh"
+
+    def testName =
+        testDirectory
+        | replace `^${quote here}/` ""
+
+    def readFile list = match list
+        Some file =
+            require Pass content = read file
+            Pass (Some content)
+        None =
+            Pass None
+
+    require Pass expectedStdout =
+        source "{testDirectory}/stdout"
+        | getPass
+        | readFile
+
+    require Pass expectedStderr =
+        source "{testDirectory}/stderr"
+        | getPass
+        | readFile
+
+    require Pass visibleFiles =
+        sources testDirectory `.*`
+
+    def inTestDir path =
+        relative testDirectory path
+
+    require Pass (Pair wakeBin wakeVisible) =
+        wakeToTest Unit
+
+    require Pass wakeUnitBin =
+        wakeUnitToTest Unit
+
+    def testJob =
+        makeExecPlan ("./{testScript.getPathName.inTestDir}", wakeUnitBin.inTestDir, Nil) (visibleFiles ++ wakeVisible)
+        | setPlanDirectory testDirectory
+        | setPlanLabel "testing: {testName}"
+        | setPlanLocalOnly True # On OS/X you cannot mount fuse within fuse
+        | setPlanStdout logNever
+        | setPlanStderr logNever
+        | runJob
+
+    require Pass jobStdout =
+        testJob.getJobStdout
+        | rmap (replace `\r` "")
```

Comment:
A named helper function like `removeCarriageReturns` could improve readability, as its used in a few places